erpnext/hr/report/salary_register/salary_register.py

```diff
@@ -74,13 +74,23 @@ def get_columns(salary_slips):
 def get_salary_slips(filters):
 	filters.update({"from_date": filters.get("from_date"), "to_date":filters.get("to_date")})
 	conditions, filters = get_conditions(filters)
-	salary_slips = frappe.db.sql("""select * from `tabSalary Slip` where docstatus = 1 %s
+	salary_slips = frappe.db.sql("""select * from `tabSalary Slip` where %s
 		order by employee""" % conditions, filters, as_dict=1)
 
 	return salary_slips or []
 
 def get_conditions(filters):
 	conditions = ""
+
+	if filters.get("docstatus"):
```

Comment:
Maybe we can skip the if-elif-elif block ?
```
doc_status = {"Draft": 0, "Submitted": 1, "Cancelled": 2}

if filters.get("docstatus"):
	conditions += "docstatus = {0}".format(doc_status(filter.get("docstatus")))
```

Reply:
Tried to use the suggested code and got the below error. 

conditions += "docstatus = {0}".format(doc_status(filter.get("docstatus")))
AttributeError: 'builtin_function_or_method' object has no attribute 'get'

I think we will have to loop through the dict after getting the filter value and then use the value in format? Is that better than if-elif-elif?

Reply:
My bad, it should be filters.get("docstatus") - missed an s there

Reply:
conditions += "docstatus = {0}".format(doc_status(filters.get("docstatus")))
TypeError: 'dict' object is not callable